libraries/AC_PID/AC_PID.h

```diff
@@ -9,102 +9,128 @@
+    float get_pi() const;
+    float get_p() const;
+    float get_i() const;
+    float get_d() const;
+    float get_ff();
+
+    // todo: remove function when it is no longer used.
+    float get_ff(float target);
 
     // reset_I - reset the integrator
-    void        reset_I();
+    void reset_I();
 
     // reset_filter - input filter will be reset to the next value provided to set_input()
-    void        reset_filter() { _flags._reset_filter = true; }
+    void reset_filter() {
+        _flags._reset_filter = true;
+    }
 
     // load gain from eeprom
-    void        load_gains();
+    void load_gains();
 
     // save gain to eeprom
-    void        save_gains();
+    void save_gains();
 
     /// operator function call for easy initialisation
-    void operator() (float p, float i, float d, float imaxval, float input_filt_hz, float dt, float ffval = 0);
+    void operator()(float p_val, float i_val, float d_val, float ff_val, float imax_val, float input_filt_T_hz, float input_filt_E_hz, float input_filt_D_hz, float dt);
 
     // get accessors
-    AP_Float   &kP() { return _kp; }
-    AP_Float   &kI() { return _ki; }
-    AP_Float   &kD() { return _kd; }
-    AP_Float   &filt_hz() { return _filt_hz; }
-    float       imax() const { return _imax.get(); }
-    float       get_filt_alpha() const;
-    float       ff() const { return _ff.get(); }
+    AP_Float &kP() { return _kp; }
+    AP_Float &kI() { return _ki; }
+    AP_Float &kD() { return _kd; }
+    AP_Float &ff() { return _kff;}
+    AP_Float &filt_T_hz() { return _filt_T_hz; }
+    AP_Float &filt_E_hz() { return _filt_E_hz; }
+    AP_Float &filt_D_hz() { return _filt_D_hz; }
+    float imax() const { return _kimax.get(); }
+    float get_filt_alpha(float filt_hz) const;
+    float get_filt_T_alpha() const;
+    float get_filt_E_alpha() const;
+    float get_filt_D_alpha() const;
 
     // set accessors
-    void        kP(const float v) { _kp.set(v); }
-    void        kI(const float v) { _ki.set(v); }
-    void        kD(const float v) { _kd.set(v); }
-    void        imax(const float v) { _imax.set(fabsf(v)); }
-    void        filt_hz(const float v);
-    void        ff(const float v) { _ff.set(v); }
-
-    float       get_integrator() const { return _integrator; }
-    void        set_integrator(float i) { _integrator = i; }
+    void kP(const float v) { _kp.set(v); }
+    void kI(const float v) { _ki.set(v); }
+    void kD(const float v) { _kd.set(v); }
+    void ff(const float v) { _kff.set(v); }
+    void imax(const float v) { _kimax.set(fabsf(v)); }
+    void filt_T_hz(const float v);
+    void filt_E_hz(const float v);
+    void filt_D_hz(const float v);
 
     // set the desired and actual rates (for logging purposes)
-    void        set_desired_rate(float desired) { _pid_info.desired = desired; }
-    void        set_actual_rate(float actual) { _pid_info.actual = actual; }
+    void set_desired_rate(float desired) { _pid_info.target = desired; }
```

Comment:
we seem to be some mixing of "desired" and "target".. not sure if we can consolidate that or not.  Not a blocker but just saying we've introduced a difference with this PR by renaming "desired" to "target" in the PID info but not in other places.

Reply:
done